File: #0007.reverse-integer.cpp
class Solution {
public:
    int reverse(int x) {

        int result = 0;
        int remainder;
        while (x) {
            remainder = x % 10;
            x /= 10;
            
            if (result > INT_MAX / 10 || (result == INT_MAX / 10 && remainder > 7)) {
                return 0;
            }
                
            if (result < INT_MIN / 10 || (result == INT_MIN / 10 && remainder < -8)) {
                return 0;
            }
            
            result = result * 10 + remainder;
        }
        
        return result;
    }
};

File: #0007.reverse-integer.cs
public class Solution {
    public int Reverse(int x) {

        long result = 0;
        while (x != 0)
        {
            result = result * 10 + (x % 10);
            x /= 10;
        }
 
        return Math.Abs(result) > Int32.MaxValue ? 0 : (int)result;
    }
}
